src/main/java/games/strategy/engine/data/GameDataMemento.java

```diff
@@ -101,11 +101,16 @@ private GameDataMemento() {}
 
     @Override
     public void exportProperties(final GameData gameData, final Map<String, Object> propertiesByName) {
-      propertiesByName.put(PropertyNames.DICE_SIDES, gameData.getDiceSides());
```

Comment:
I'd put this line inside the try block.
In case this method throws an exception for whatever reason but still aquires the lock, the lock should be released properly